colour/utilities/array.py

```diff
@@ -1906,13 +1912,27 @@ def interval(distribution: ArrayLike, unique: bool = True) -> NDArray:
     """
 
     distribution = as_float_array(distribution)
+    hash_key = hash(
+        (
+            distribution.tobytes(),
+            np.array(distribution.shape).tobytes(),
+            unique,
```

Comment:
**Discussion/Minor**

I wish Numpy offered a native way to hash ndarrays, given we do that often, we should maybe write a small wrapper function to do that.

`shape` is a tuple btw, so you don't need to convert it:

```
>>> hash((1, 2, 3))
529344067295497451
```